### PR TITLE
Fixes Jenkins run_sphinx job (docs) -  FLOC-2857

### PR DESCRIPTION
**Pull request by Azulinho**
This PR fixes the issue where the run_sphinx job was failling.
http://ci-live.clusterhq.com:8080/job/ClusterHQ-flocker/job/master/job/run_sphinx/

And enables the uploads of the resulting html files to an S3 bucket on AWS.
http://ci-live.clusterhq.com:8080/job/ClusterHQ-flocker/job/jenkins-docs-FLOC-2857/job/run_sphinx/8/console

[Review on Reviewable](https://reviewable.io/reviews/clusterhq/flocker/2115)

### CODE REVIEW

File: build.yaml [hunk 1]
```diff
@@ -335,22 +335,12 @@ common_cli:
...
     parse_logs sphinx-build -d _build/doctree -b linkcheck . _build/linkcheck
-    let status=status+\$?
     # build html pages
     parse_logs sphinx-build -d _build/doctree -b html . _build/html
-    exit \${status}
-    # TODO:
-    # upload html
-    #link-release-documentation
-    #upload-release-documentation
-    cd -
```
Comment (Azulinho) [inline build.yaml]: we don't need any of this stuff

File: build.yaml [hunk 2]
```diff
@@ -717,6 +707,23 @@ common_cli:
...
+  install_aws_cli: &install_aws_cli |
+    # installs the AWS python based cli
+    # we use the aws cli to upload the docs files to S3
+    \${PARSE_LOGS}  pip install awscli \${PIP_ADDITIONAL_OPTIONS}
+
+  get_s3_creds_for_bucket_staging_docs: &get_s3_creds_for_bucket_staging_docs |
+    # collects the S3 credentials that allows us to upload files to the S3
+    # staging bucket for our docs
+    . /tmp/s3_staging_docs_clusterhq_com.sh
+
```
Comment (Azulinho) [inline build.yaml]: the file with the credentials are deployed by the jenkins master to the slave when the slave is instantiated.

File: build.yaml [hunk 3]
```diff
@@ -928,14 +935,16 @@ job_type:
   # http://build.clusterhq.com/builders/flocker-docs
   run_sphinx:
     run_sphinx:
-      on_nodes_with_labels: 'aws-centos-7-T2Small'
+      on_nodes_with_labels: 'aws-ubuntu-trusty-T2Small'
```
Comment (Azulinho) [inline build.yaml]: this is the reason the job has been failling so far, something is fishy with sphinx on centos boxes.
It always fails on Centos, but works without any changes on Ubuntu

File: build.yaml [hunk 4]
```diff
@@ -928,14 +935,16 @@ job_type:
...
         - { type: 'shell',
             cli: [ *hashbang, *add_shell_functions,  *setup_pip_cache,
                    *cleanup, *setup_venv, *setup_flocker_modules,
-                   *setup_aws_env_vars, *run_sphinx ]
+                   *install_aws_cli, *run_sphinx,
+                   *get_s3_creds_for_bucket_staging_docs,
+                   *upload_new_docs_html_to_s3_staging_docs]
           }
-      timeout: 30
+      timeout: 10
```
Comment (Azulinho) [inline build.yaml]: the job only takes 3 minutes to run, so not point in defining a timeout 10 times longer than that.

File: build.yaml (hunk 1 above)
Comment (jml) [inline build.yaml]: Then delete it?

File: build.yaml (hunk 2 above)
Comment (jml) [inline build.yaml]: That comment seems relevant to the code, not just to the review. Please add it as a comment in the yaml file.

File: build.yaml (hunk 4 above)
Comment (jml) [inline build.yaml]: Strongly agree.

**Comment by jml** [comment]
Looks good. Nice seeing these jobs get added.

Please,
* Delete some commented out code
* Add one of the review comments to the `build.yaml` file itself

Then merge.

File: build.yaml (hunk 2 above)
Comment (Azulinho) [inline build.yaml]: +1

File: build.yaml (hunk 1 above)
Comment (Azulinho) [inline build.yaml]: there is a '-' there, so its a git deletion

**Comment by Azulinho** [comment]
@jml cookies towards your way

File: build.yaml (hunk 1 above)
Comment (jml) [inline build.yaml]: Oh, duh. My bad.

**Pull request merged**